server/filestore.go

```diff
@@ -4080,16 +4083,16 @@ func (fs *fileStore) loadLast(subj string, sm *StoreMsg) (lsm *StoreMsg, err err
 
 // LoadLastMsg will return the last message we have that matches a given subject.
 // The subject can be a wildcard.
-func (fs *fileStore) LoadLastMsg(subject string, sm *StoreMsg) (*StoreMsg, error) {
+func (fs *fileStore) LoadLastMsg(subject string, smv *StoreMsg) (sm *StoreMsg, err error) {
 	if subject == _EMPTY_ || subject == fwcs {
-		sm, _ = fs.msgForSeq(fs.lastSeq(), sm)
+		sm, err = fs.msgForSeq(fs.lastSeq(), sm)
 	} else {
-		sm, _ = fs.loadLast(subject, sm)
+		sm, err = fs.loadLast(subject, sm)
 	}
-	if sm == nil {
-		return nil, ErrStoreMsgNotFound
+	if err != nil && err != ErrStoreClosed {
```

Comment:
This is causing a panic on some client tests and that is because the low level fs.loadLast() may not find the message but not have an error, so `sm` will be nil. The original code as you can see above was checking for `sm` nil and in that case return `ErrStoreMsgNotFound`. We still need that or then make that test in fs.loadLast().

Reply:
Wil lfix.